flake.nix

```diff
@@ -0,0 +1,448 @@
+        # LLVM code coverage
+        craneLLVMLib = (crane.mkLib pkgs).overrideToolchain rustLLVMTarget;
+        craneNightlyLib = (crane.mkLib pkgs).overrideToolchain rustNightlyTarget;
+
+        # Common inputs for all derivations
+        buildInputs = [
+          # Add additional build inputs here
+          pkgs-bitcoind.bitcoind
+          pkgs-kitman.esplora
+          pkgs.openssl
+          pkgs.openssl.dev
+          pkgs.pkg-config
+          pkgs.curl
+          pkgs.libiconv
+        ] ++ lib.optionals isDarwin libsDarwin;
+
+        # WASM deps
+        WASMInputs = [
+          # Additional wasm specific inputs can be set here
+          pkgs.llvmPackages_14.clang-unwrapped
+          pkgs.llvmPackages_14.stdenv
+          pkgs.llvmPackages_14.libcxxClang
+          pkgs.llvmPackages_14.libcxxStdenv
+        ];
+
+        nativeBuildInputs = [
+          # Add additional build inputs here
+          pkgs.python3
+        ] ++ lib.optionals isDarwin [
+          # Additional darwin specific native inputs can be set here
+        ];
+
+        # Common derivation arguments used for all builds
+        commonArgs = {
+          # When filtering sources, we want to allow assets other than .rs files
+          src = lib.cleanSourceWith {
+            src = ./.; # The original, unfiltered source
+            filter = path: type:
+              # esplora uses `.md` in the source code
+              (lib.hasSuffix "\.md" path) ||
+              # bitcoind_rpc uses `.db` in the source code
+              (lib.hasSuffix "\.db" path) ||
+              # Default filter from crane (allow .rs files)
+              (craneLib.filterCargoSources path type);
+          };
+
+          # Fixing name/version here to avoid warnings
+          # This does not interact with the versioning
+          # in any of bdk crates' Cargo.toml
+          pname = "crates";
+          version = "0.1.0";
+
+          inherit buildInputs;
+          inherit nativeBuildInputs;
+          # Additional environment variables can be set directly
+          BITCOIND_EXEC = "${pkgs.bitcoind}/bin/bitcoind";
+          ELECTRS_EXEC = "${pkgs-kitman.esplora}/bin/esplora";
+        };
+
+        # MSRV derivation arguments
+        MSRVArgs = {
+          cargoLock = ./CargoMSRV.lock;
+        };
+
+        # WASM derivation arguments
+        WASMArgs = {
+          CARGO_BUILD_TARGET = "wasm32-unknown-unknown";
+          buildInputs = buildInputs ++ WASMInputs;
+          inherit nativeBuildInputs;
+          # crane tries to run the WASM file as if it were a binary
+          doCheck = false;
+          # just build bdk for now
+          cargoExtraArgs = "--locked --target wasm32-unknown-unknown -p bdk --no-default-features --features bitcoin/no-std,miniscript/no-std,bdk_chain/hashbrown,dev-getrandom-wasm";
+          # env vars
+          CC = "${stdenv.cc.nativePrefix}cc";
```

Comment:
**suggestion:** Please verify if this even works, because IIRC `stdenv.mkDerivation` overrides `CC` and `CXX` anyway. If you want these changed, you should override `stdenv` https://github.com/ipetkov/crane/blob/master/docs/API.md#optional-attributes-15 . Again, IIRC.